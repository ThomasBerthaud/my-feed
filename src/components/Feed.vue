<template>
  <div class="tile is-parent">
    <article class="tile is-child box has-text-centered">
      <p class="title">{{ feed.title }}</p>
      <div v-html="feed.summary"></div>
      <router-link :to="'/details/' + feed.id" class="button is-text">
        Voir les autres...
      </router-link>
    </article>
  </div>
</template>

<script>
import axios from "axios";
export default {
  name: "Feed",
  props: {
    feedParams: {
      type: Object,
      required: true
    }
  },
  data() {
    return {
      feed: {}
    };
  },
  created() {
    axios.get(`/feed/latest?url=${this.feedParams.url}&type=${this.feedParams.type}`).then(response => {
      this.feed = response.data;
      console.log(this.feed);
    });
  }
};
</script>

<style></style>
