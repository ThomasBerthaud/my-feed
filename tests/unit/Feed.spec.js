import axios from "axios";
import { shallowMount } from "@vue/test-utils";
import Feed from "@/components/Feed.vue";

jest.mock("axios");

describe("Feed.vue", () => {
  it("displays feed card after http call", async () => {
    // GIVEN
    const feedUrl = "feedUrl";
    const response = { id: "feedID", title: "title", entries: [{ summary: "feed summary" }] };
    axios.get.mockResolvedValue({ data: response });

    // WHEN
    const wrapper = shallowMount(Feed, {
      propsData: { feedUrl },
      stubs: ["router-link", "router-view"]
    });

    // THEN
    await wrapper.vm.$nextTick();
    expect(axios.get.mock.calls[0]).toContain("/feed/latest?url=feedUrl");
    expect(wrapper.element).toMatchSnapshot();
  });
});
