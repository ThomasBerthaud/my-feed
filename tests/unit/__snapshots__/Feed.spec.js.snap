// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed.vue displays feed card after http call 1`] = `
<div>
  <div
    class="card"
  >
    <header
      class="card-header"
    >
      <p
        class="card-header-title"
      >
        title
      </p>
       
      <router-link-stub
        aria-label="feed settings"
        class="card-header-icon"
        to="/settings/feedID"
      >
        <span
          class="icon"
        >
          <i
            class="fas fa-cog"
          />
        </span>
      </router-link-stub>
    </header>
     
    <div
      class="card-content limit-size"
    >
      <div
        class="content"
      >
        <div>
          feed summary
        </div>
         
        <time>
          
        </time>
      </div>
    </div>
     
    <div
      class="card-footer"
    >
      <router-link-stub
        class="card-footer-item"
        to="/details/feedID"
      >
        Voir les autres...
      </router-link-stub>
       
      <a
        aria-label="external link"
        class="card-footer-item"
      >
        <span>
          Acceder au site
        </span>
         
        <span
          class="icon"
        >
          <i
            class="fas fa-external-link-alt"
          />
        </span>
      </a>
    </div>
  </div>
   
  <!---->
</div>
`;
